functional_tests/magic/02-no-keras.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+"""magic test with no ml modules installed.
+
+---
+id: 0.magic.02-no-keras
+plugin:
+  - wandb
+tag:
+  shard: noml
+command:
+    args:
+        - --layers=5
+assert:
+  - :wandb:runs_len: 1
+  - :wandb:runs[0][config][epochs]: 4
+  - :wandb:runs[0][config][layers]: 5
+  - :wandb:runs[0][exitcode]: 0
+"""
```

Comment:
is there anything else `magic` does if no ml is installed?

Reply:
i guess i could check stdout/stderr.  but that is it.   Will add more magic tests when I get time, though looking at this code it is clear it needs some serious cleanups (and annotations!) :)

Reply:
yeah... I was looking at this code and i think it needs a lot of love. 
Also I see that we are using `global` a lot in this file.